velocity-custom-parser-example/pom.xml

```diff
@@ -0,0 +1,287 @@
+                        <goals>
+                            <goal>clean</goal>
+                        </goals>
+                        <configuration>
+                            <excludeDefaultDirectories>true</excludeDefaultDirectories>
+                            <filesets>
+                                <fileset>
+                                    <directory>${project.build.directory}/generated-sources/javacc/</directory>
+                                    <includes>
+                                        <include>**/*.java</include>
+                                    </includes>
+                                    <excludes>
+                                        <exclude>**/*${parser.basename}*.java</exclude>
+                                    </excludes>
+                                </fileset>
+                                <fileset>
+                                    <directory>${project.build.directory}/generated-sources/jjtree/</directory>
+                                    <includes>
+                                        <include>**/node/*.java</include>
+                                    </includes>
+                                    <excludes>
+                                        <exclude>**/node/*${parser.basename}*.java</exclude>
+                                    </excludes>
+                                </fileset>
+                            </filesets>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!-- add missing imports to some parser generated files -->
+            <plugin>
+                <groupId>com.google.code.maven-replacer-plugin</groupId>
+                <artifactId>replacer</artifactId>
+                <executions>
+                    <execution>
+                        <id>patch-parser-files</id>
+                        <phase>process-sources</phase>
+                        <goals>
+                            <goal>replace</goal>
+                        </goals>
+                        <configuration>
+                            <includes>
+                                <include>${project.build.directory}/generated-sources/jjtree/**/JJT${parser.basename}ParserState.java</include>
+                                <include>${project.build.directory}/generated-sources/jjtree/**/${parser.basename}ParserVisitor.java</include>
+                            </includes>
+                            <replacements>
+                                <replacement>
+                                    <token>import ${parser.package}.*;</token>
+                                    <value>import ${parser.package}.*;
+import org.apache.velocity.runtime.parser.node.*;</value>
+                                </replacement>
+                            </replacements>
+                        </configuration>
+                    </execution>
```

Comment:
All of this look unbelieveably clumsy to me, I guess there is no better way at the moment to solve this. It has so many hoops to get custom parser that most people will deem this unusable :-(

Reply:
I don't understand your point. We'll document the process on the site (it boils down to copy paste the example pom and changing the parser properties at the top), yes there are several steps to deal with javacc, made a bit verbose by the maven API, but it stays quite straightforward. After all, we're speaking of an example.

Reply:
Agreed.